src/check/property/SkipAfterProperty.ts

```diff
@@ -17,7 +17,12 @@ export class SkipAfterProperty<Ts> implements IProperty<Ts> {
   generate = (mrng: Random, runId?: number) => this.property.generate(mrng, runId);
   run = (v: Ts) => {
     if (this.getTime() >= this.skipAfterTime) {
-      return new PreconditionFailure(this.interruptExecution);
+      const preconditionFailure = new PreconditionFailure(this.interruptExecution);
+      if (this.isAsync()) {
+        return Promise.resolve(preconditionFailure) as any; // IsAsync => Promise<PreconditionFailure | string | null>
+      } else {
+        return preconditionFailure as any; // !IsAsync => PreconditionFailure | string | null
+      }

```

Comment:
I need to think about it. The best solution might just be to say that we return a `IRawProperty<Ts>` with no more details on the sync vs async behaviour. 🤔